### PR TITLE
Re-parameterization and better hyper-parameters for Data Re-uploading example

### PR DESCRIPTION
**Pull request by quantshah**
Hi @co9olguy @mariaschuld @josh146, the uploaded tutorial looked very nice apart from the final decision boundary which was not completely circular. I found some better hyper parameters which consistently give a nicer decision boundary.

Figure_1

Also I realised that the since we are not strictly trying to reproduce the work by Peres et. al., I do not need to parameterise the inputs as U(wx +b) as they do and neither am I using the same fidelity function etc. so it is better to stick to the explanation in the tutorial (alternate layers of data and weight unitaries). This makes it consistent.

### CODE REVIEW

**Comment by josh146** [comment]
Thanks @quantshah! Something that's worth considering is making this tutorial deterministic by setting the random seed

**Comment by mariaschuld** [comment]
Looks good, seconded @josh146 's comment!

**Comment by quantshah** [comment]
Thanks for the suggestion @josh146 . I added a `np.random.seed(42)` in the beginning of the code. Re-running the example with this seed gets pretty consistent results and a nice decision boundary. If this looks good then please review and merge this PR to update the rendered notebooks.

File: examples/pennylane_run_data_reuploading_classifier.py [hunk 1]
```diff
@@ -241,25 +245,26 @@ def density_matrix(state):
 
 ##############################################################################
 # Simple classifier with data reloading and fidelity loss
-# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+##############################################################################
```
Comment (josh146) [inline examples/pennylane_run_data_reuploading_classifier.py]: What is the reason for this change here? Due to the way ReST does section headings, it is context
dependent. Since neither `~` nor `#` has been used previously as a section heading, this should have no
functional difference. `~` is preferred, however, to stick with prior PL docs convention.

Also, to keep it in line with ReST formatting, the underline should match the length of the section heading.

---

Unless this is no longer meant to be a ReST section heading, and the intention is to downgrade it to a comment?

File: examples/pennylane_run_data_reuploading_classifier.py [hunk 2]
```diff
@@ -241,25 +245,26 @@ def density_matrix(state):
...
 
 
 @qml.qnode(dev)
-def qcircuit(var, x=None, y=None):
+def qcircuit(params, x=None, y=None):
     """A variational quantum circuit representing the Universal classifier.
 
     Args:
-        var (array[float]): array of variables
+        params (array[float]): array of parameters
```
Comment (josh146) [inline examples/pennylane_run_data_reuploading_classifier.py]: :+1:

File: examples/pennylane_run_data_reuploading_classifier.py [hunk 3]
```diff
@@ -293,21 +298,20 @@ def cost(params, x, y, state_labels=None):
...
-        loss = loss + (1 - f) ** 2
+        f = qcircuit(params, x=x[i], y=dm_labels[y[i]])
+        loss = loss + (1 - f)**2
     return loss / len(x)
 
 
 ##############################################################################
 # Utility functions for testing and creating batches
-# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+##############################################################################
```
Comment (josh146) [inline examples/pennylane_run_data_reuploading_classifier.py]: same question here wrt ReST section headings

File: examples/pennylane_run_data_reuploading_classifier.py [hunk 4]
```diff
@@ -378,8 +381,8 @@ def iterate_minibatches(inputs, targets, batch_size):
 
 
 ##############################################################################
-# Training on circle dataset
-# ~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Training a classifier on the circle dataset
+##############################################################################
```
Comment (josh146) [inline examples/pennylane_run_data_reuploading_classifier.py]: same question here wrt ReST section headings

File: examples/pennylane_run_data_reuploading_classifier.py [hunk 5]
```diff
@@ -241,25 +245,26 @@ def density_matrix(state):
...
         x (array[float]): single input vector
         y (array[float]): single output state density matrix
 
     Returns:
         float: fidelity between output state and input
     """
-    for v in var:
-        qml.Rot(*v, wires=0)
+    for p in params:
+        qml.Rot(*x, wires=0)
```
Comment (co9olguy) [inline examples/pennylane_run_data_reuploading_classifier.py]: Was the input vector not being considered before??

**Comment by josh146** [comment]
@quantshah: ah, I see. Note that all tutorials can be downloaded as executable python scripts, so probably don't need to worry too much about copying and pasting (I personally find it easier to read when the code-blocks are broken up).

As an aside: all ReST-blocks start with
```
##############################################################################
```
so your section headings will continue to be ReST formatted. You would need to remove the initial 79 hashes (or change the symbol) to avoid it rendering.

**Comment by quantshah** [comment]
@josh146 I decided to use the RST blocks and also print the results better by mentioning the weights at each layer. Please review and merge at will. :-)
